src/poc/miner_poc_grpc_client_statem.erl

```diff
@@ -611,10 +622,10 @@ do_send_report(Req, ReportType, Report, OnionKeyHash, Connection, RetryAttempts)
                 {ok, _} ->
                     ok;
                 _ ->
-                    ?MODULE:send_report(ReportType, Report, OnionKeyHash, RetryAttempts - 1)
+                    ?MODULE:retry_report(ReportType, Report, OnionKeyHash, RetryAttempts - 1)
```

Comment:
Why not just put a `timer:sleep/1` in here and then call the existing `send_report/4` function?

Reply:
If you’re looking to get the gen_statem out of this function and not sit in the timer then I’d have this guy schedule the `send_after` and interpose a small handle_info that just catches that message and then calls `send_report/4` passing down the args and decrementing the retry. Mimicking a cast in a simple send message with the magic `'$gen_cast'` atom seems brittle to me

Reply:
that's reasonable